nsxt/resource_nsxt_vpc_service_profile.go

```diff
@@ -0,0 +1,445 @@
+/* Copyright © 2024 Broadcom, Inc. All Rights Reserved.
+   SPDX-License-Identifier: MPL-2.0 */
+
+package nsxt
+
+import (
+	"fmt"
+	"log"
+	"reflect"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/protocol/client"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/model"
+	clientLayer "github.com/vmware/vsphere-automation-sdk-go/services/nsxt/orgs/projects"
+
+	utl "github.com/vmware/terraform-provider-nsxt/api/utl"
+	"github.com/vmware/terraform-provider-nsxt/nsxt/metadata"
+)
+
+var vpcServiceProfileModeValues = []string{
+	model.VpcProfileDhcpConfig_MODE_IP_ALLOCATION_BY_PORT,
+	model.VpcProfileDhcpConfig_MODE_IP_ALLOCATION_BY_MAC,
+	model.VpcProfileDhcpConfig_MODE_RELAY,
+	model.VpcProfileDhcpConfig_MODE_DEACTIVATED,
+}
+
+var vpcServiceProfileLogLevelValues = []string{
+	model.PolicyVpcDnsForwarder_LOG_LEVEL_DEBUG,
+	model.PolicyVpcDnsForwarder_LOG_LEVEL_INFO,
+	model.PolicyVpcDnsForwarder_LOG_LEVEL_ERROR,
+	model.PolicyVpcDnsForwarder_LOG_LEVEL_WARNING,
+	model.PolicyVpcDnsForwarder_LOG_LEVEL_FATAL,
+}
+
+var vpcServiceProfileSchema = map[string]*metadata.ExtendedSchema{
+	"nsx_id":       metadata.GetExtendedSchema(getNsxIDSchema()),
+	"path":         metadata.GetExtendedSchema(getPathSchema()),
+	"display_name": metadata.GetExtendedSchema(getDisplayNameSchema()),
+	"description":  metadata.GetExtendedSchema(getDescriptionSchema()),
+	"revision":     metadata.GetExtendedSchema(getRevisionSchema()),
+	"tag":          metadata.GetExtendedSchema(getTagsSchema()),
+	"context":      metadata.GetExtendedSchema(getContextSchema(true, false, false)),
+	"is_default": {
+		Schema: schema.Schema{
+			Type:     schema.TypeBool,
+			Optional: true,
+		},
+		Metadata: metadata.Metadata{
+			SchemaType:   "bool",
+			SdkFieldName: "IsDefault",
+		},
+	},
+	"mac_discovery_profile": {
```

Comment:
Various profiles are policy paths? If so, they can be validated with `validatePolicyPath()`